src/image.js

```diff
@@ -0,0 +1,86 @@
+/**
+ * @license Copyright (c) 2003-2016, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+import Feature from '../core/feature.js';
+import ImageEngine from './imageengine.js';
+import ModelRange from '../engine/model/range.js';
+import MouseDownObserver from './mousedownobserver.js';
+
+/**
+ * The image feature.
+ *
+ * Uses {@link image.ImageEngine}.
+ *
+ * @memberOf image
+ * @extends core.Feature
+ */
+export default class Image extends Feature {
+	/**
+	 * @inheritDoc
+	 */
+	static get requires() {
+		return [ ImageEngine ];
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	init() {
+		const viewDocument = this.editor.editing.view;
+
+		// If mouse down is pressed create selection over whole view element.
+		viewDocument.addObserver( MouseDownObserver );
+		this.listenTo( viewDocument, 'mousedown', this._onMouseDown, { context: this } );
```

Comment:
We don't use `context` option (it will be removed). Use `() => this._onMouseDown()`.